src/main/java/com/overops/plugins/sonar/rules/checks/OverOpsBaseExceptionCheck.java

```diff
@@ -0,0 +1,127 @@
+package com.overops.plugins.sonar.rules.checks;
+
+import com.overops.plugins.sonar.measures.OverOpsEventsStatistic;
+import com.overops.plugins.sonar.measures.OverOpsMetrics;
+import com.takipi.api.client.functions.output.RegressionRow;
+import com.takipi.api.client.result.event.EventResult;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+
+import javax.annotation.Nonnull;
+import java.io.File;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.overops.plugins.sonar.OverOpsPlugin.serviceId;
+import static com.overops.plugins.sonar.OverOpsPlugin.overOpsEventsStatistic;
+import static com.overops.plugins.sonar.measures.OverOpsQualityGateStat.INCREASING_QG_MARKER;
+
+public abstract class OverOpsBaseExceptionCheck extends BaseTreeVisitor implements JavaFileScanner {
+    private static final Logger log = Loggers.get(OverOpsBaseExceptionCheck.class);
+    private JavaFileScannerContext context;
+    private File file;
+    protected OverOpsMetrics.OverOpsMetric metric;
+    public static Map<String, Long> fileToLinesCount = new HashMap<>();
+
+    @Override
+    public void scanFile(final @Nonnull JavaFileScannerContext ctx) {
+        if (overOpsEventsStatistic == null) {
+            return;
+        }
+
+        context = ctx;
+        file = context.getFile();
+        String filePathJavaStyle = file.getAbsolutePath().replaceAll("/", ".");
+        List<OverOpsEventsStatistic.ClassStat> statForThisFile = overOpsEventsStatistic.getStatistic()
```

Comment:
please do some code cleanup here and perhaps break this to some methods that actually explains what is being done

Reply:
done